pkg/core/resources/apis/mesh/proxytemplate_validator.go

```diff
@@ -29,10 +34,142 @@ func (t *ProxyTemplateResource) Validate() error {
+		if err := ValidateResourceYAMLPatch(&envoy_api.Listener{}, listenerMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	case mesh_proto.OpRemove:
+	default:
+		verr.AddViolation("operation", fmt.Sprintf("invalid operation. Available operations: %q, %q, %q", mesh_proto.OpAdd, mesh_proto.OpPatch, mesh_proto.OpRemove))
+	}
+	return verr
+}
+
+func validateClusterModification(clusterMod *mesh_proto.ProxyTemplate_Modifications_Cluster) validators.ValidationError {
+	verr := validators.ValidationError{}
+	switch clusterMod.Operation {
+	case mesh_proto.OpAdd:
+		if clusterMod.Match != nil {
+			verr.AddViolation("match", "cannot be defined")
+		}
+		if err := ValidateResourceYAML(&envoy_api.Cluster{}, clusterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	case mesh_proto.OpPatch:
+		if err := ValidateResourceYAMLPatch(&envoy_api.Cluster{}, clusterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	case mesh_proto.OpRemove:
+	default:
+		verr.AddViolation("operation", fmt.Sprintf("invalid operation. Available operations: %q, %q, %q", mesh_proto.OpAdd, mesh_proto.OpPatch, mesh_proto.OpRemove))
+	}
+	return verr
+}
+
+func validateNetworkFilterModification(networkFilterMod *mesh_proto.ProxyTemplate_Modifications_NetworkFilter) validators.ValidationError {
+	verr := validators.ValidationError{}
+	validateResource := func() {
+		if err := ValidateResourceYAML(&envoy_listener.Filter{}, networkFilterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	}
+	switch networkFilterMod.Operation {
+	case mesh_proto.OpAddFirst:
+		validateResource()
+	case mesh_proto.OpAddLast:
+		validateResource()
+	case mesh_proto.OpAddBefore:
+		if networkFilterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty. You need to pick a filter before which this one will be added")
+		}
+		validateResource()
+	case mesh_proto.OpAddAfter:
+		if networkFilterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty. You need to pick a filter after which this one will be added")
+		}
+		validateResource()
+	case mesh_proto.OpPatch:
+		if networkFilterMod.GetMatch().GetName() == "" {
+			verr.AddViolation("match.name", "cannot be empty")
+		}
+		if err := ValidateResourceYAMLPatch(&envoy_listener.Filter{}, networkFilterMod.Value); err != nil {
+			verr.AddViolation("value", fmt.Sprintf("native Envoy resource is not valid: %s", err.Error()))
+		}
+	case mesh_proto.OpRemove:
+	default:
+		verr.AddViolation("operation", fmt.Sprintf("invalid operation. Available operations: %q, %q, %q, %q, %q", mesh_proto.OpAddFirst, mesh_proto.OpAddLast, mesh_proto.OpAddBefore, mesh_proto.OpAddAfter, mesh_proto.OpPatch))
```

Comment:
and here

Reply:
It's not missed, there is nothing to validate there :)